src/main/java/org/opensearch/knn/index/query/KNNWeight.java

```diff
@@ -145,6 +146,12 @@ public PerLeafResult searchLeaf(LeafReaderContext context, int k) throws IOExcep
             Map<Integer, Float> result = doExactSearch(context, new BitSetIterator(filterBitSet, cardinality), cardinality, k);
             return new PerLeafResult(filterWeight == null ? null : filterBitSet, result);
         }
+        /*
+         * If filters match all docs in this segment, then there is no need to do any extra step
```

Comment:
Could you give more context to the comment?? Would like to see why there's no need for extra steps for all doc matches case, and what passing `new FixedBitSet(0)` means. Basically, we want to save a bitset look up cost when it's not required as possible.